File: src/StripController/stripper/SetPixelColorCommand.cpp
#include "SetPixelColorCommand.h"

SetPixelColorCommand::SetPixelColorCommand(Strip* strip, Logger* logger)
	:_strip(strip), _logger(logger)
{
}

int SetPixelColorCommand::GetCommandId()
{
	return Command::SetPixelColorCommandId;
}

void SetPixelColorCommand::Handle(byte* command)
{
	_logger->Trace("Handle SetPixelColorCommand begin");

	int offset = 1;

	byte pixel = command[offset++];
	byte r = command[offset++];
	byte g = command[offset++];
	byte b = command[offset++];

	char buffer[255];
	sprintf(buffer, "Arguments: {\"pixel\": %d, \"r\": %d, \"g\": %d, \"b\": %d}", pixel, r, g, b);
	_logger->Trace(buffer);

	_strip->SetPixelColor(pixel, r, g, b);

	_logger->Trace("Handle SetPixelColorCommand end");
}

File: src/StripController/StripController.Infrastructure/StripWrapper/SetPixelColorCommand.cs
namespace StripController.Infrastructure.StripWrapper
{
    class SetPixelColorCommand : Command
    {
        private readonly byte _pixel;
        private readonly byte _r;
        private readonly byte _g;
        private readonly byte _b;

        public SetPixelColorCommand(byte pixel, byte r, byte g, byte b)
        {
            _pixel = pixel;
            _r = r;
            _g = g;
            _b = b;
        }

        public override byte[] GetBody()
        {
            return new[] {_pixel, _r, _g, _b};
        }

        public override int GetCommandId()
        {
            return Commands.SetPixelColorCommandId;
        }
    }
}
